workers/worker-performance.worker.js

```diff
@@ -0,0 +1,128 @@
+        // We want to skip over values that are 0 because of TAO securty rescritions
+        // Or else this test will fail. This can happen for "requestStart", "responseStart".
+        if (timestamp != 0) {
+            assert_greater_than_equal(timestamp, currentTimestamp, "Resource entry " + name + " is after " + currentTimestampName);
+            currentTimestamp = timestamp;
+            currentTimestampName = name;
+        }
+    });
+    assert_greater_than(entries[0].responseEnd, entries[0].startTime, "The resource request should have taken at least some time");
+     // We requested a delay of 250ms, but it could be a little bit less, and could be significantly more.
+    assert_greater_than(entries[0].responseEnd - entries[0].responseStart, 230, "Resource timing numbers reflect reality somewhat");
+}, "Resource timing seems to work in workers");
+
+test(function testPerformanceClearResourceTimings () {
+    performance.clearResourceTimings();
+    assert_equals(performance.getEntriesByType("resource").length, 0, "clearResourceTimings cleared the resource timings");
+}, "performance.clearResourceTimings in workers");
+
+test(function testPerformanceSetResourceTimingBufferSize () {
+    performance.setResourceTimingBufferSize(0);
+    var xhr = new XMLHttpRequest();
+    xhr.open("GET", "/resources/testharness.js", false);  // synchronous request
+    xhr.send();
+    assert_equals(performance.getEntriesByType("resource").length, 0, "setResourceTimingBufferSize(0) prevents resource entries from being added");
+}, "performance.setResourceTimingBufferSize in workers");
+
+test(function testPerformanceHasNoTiming () {
+    assert_equals(typeof(performance.timing), "undefined", "performance.timing is undefined");
+}, "performance.timing is not available in workers");
+
+test(function testPerformanceHasNoNavigation () {
+    assert_equals(typeof(performance.navigation), "undefined", "performance.navigation is undefined");
+}, "performance.navigation is not available in workers");
+
+test(function testPerformanceHasNoToJSON () {
+    assert_equals(typeof(performance.toJSON), "undefined", "performance.toJSON is undefined");
+}, "performance.toJSON is not available in workers");
+
+test(function testPerformanceNoNavigationEntries () {
+    assert_equals(performance.getEntriesByType("navigation").length, 0, "getEntriesByType(\"navigation\") returns nothing");
+    assert_equals(performance.getEntriesByName("document", "navigation").length, 0, "getEntriesByName(\"document\", \"navigation\") returns nothing");
+    assert_equals(performance.getEntriesByName("document").length, 0, "getEntriesByName(\"document\") returns nothing");
+    var hasNavigation = performance.getEntries().some((e,i,a) => {
+        return e.entryType == "navigation";
+    });
+    assert_false(hasNavigation, "getEntries should return no navigation entries.");
+
+}, "There are no navigation type performance entries in workers");
+
+done();
```

Comment:
Looks good. This test does a covers a lot of basic API surface, do you also want to sanity check `PerformanceObserver` in workers?